src/avx/avx32_butterflies.rs

```diff
@@ -37,9 +37,9 @@ macro_rules! boilerplate_fft_simd_butterfly {
         }
 
         impl<T: FftNum> Fft<T> for $struct_name<f32> {
-            fn process_outofplace_with_scratch(
+            fn process_immutable_with_scratch(
                 &self,
-                input: &mut [Complex<T>],
+                input: &[Complex<T>],
                 output: &mut [Complex<T>],
                 _scratch: &mut [Complex<T>],
             ) {
```

Comment:
One pitfall here is that `fft_error_outofplace` has a panic message that specifically mentions the method name `process_outofplace_with_scratch`, so using that from here will give a confusing error message.

Because of that, I sadly don't think the butterfly boilerplate macros can forward their implementations like this. I think we should make a new `fft_error_immut` function that mentions the correct function name, and call it from here, and provide a full out of place implementation that calls its own method. This goes for all of the butterfly implementations.